### ISSUE TITLE
Syntax highlighting for `$: if( ) { }`

### ISSUE THREAD

**Issue by hyunbinseo** [sveltejs/language-tools#2249]
## TL:DR;

Since the syntax highlighting works fine on VS Code, I was assuming that it was a GitHub linguist issue.

However, there was a feedback that it might be the grammar issue. Therefore, submitting this issue.

Reference https://github.com/github-linguist/linguist/issues/6673

## Describe the issue

Svelte's `$: if( ) { }` syntax seems to be broken on github.com.

- [Minimum Reproducible Example](https://gist.github.com/hyunbinseo/2b5944071cd8d37e8b0a7ac0fa2e4b2a)
- [Example file](https://github.com/sveltejs/sites/blob/master/packages/repl/src/lib/Input/ModuleEditor.svelte) with this code usage.

Currently, `lang="ts"` is required to correctly highlight this syntax.

![image](https://github.com/github-linguist/linguist/assets/47051820/151a9a0c-51a0-46ad-bf04-6a29bf6369ff)

**Comment by dummdidumm** [comment]
Given that it's somewhat of an edge case and that this syntax is going to be less used with Svelte 5+, I'm putting this in the "not that important" category. I also suspect this might have to do with the JS grammar, AFAIK we just reuse it here.

**Comment by jasonlyu123** [comment]
GitHub linguist is using a fork of a fork from an ancient version of svelte-atom. They switched to the latest Svelte atom a while ago but are having some trouble with some injections. They eventually reverted it because they couldn't fix the sunsetted internal highlighting tool they used. 

Even if they're using the grammar we're using now, and the embed problem doesn't exist. I don't think it would solve the problem. What I suspect is that they are using some old version of Atom javascript when it comes to HTML-like Textmate grammar, which doesn't support optional chaining. 

this also happens to vue
```vue
<script>
hi?.();
</script>
```

```vue
<script lang="ts">
hi?.();
</script>
```

**Comment by dummdidumm** [comment]
Closing as won't do, with Svelte 5 deprecating this syntax
